flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -702,11 +704,14 @@ private void disposeAllOperators(boolean logOnlyErrors) throws Exception {
 						operator.dispose();
 					}
 					catch (Exception e) {
-						LOG.error("Error during disposal of stream operator.", e);
+						disposalException = ExceptionUtils.firstOrSuppressed(e, disposalException);
 					}
 				}
 			}
 			disposedOperators = true;
+			if (disposalException != null) {
+				throw disposalException;
```

Comment:
So we ignore `logOnlyErrors` now and throw the exception anyways.
This prevents some other cleanup in `cleanUpInvoke`, which is not intended I guess.

Instead, we could pass "main exception" to `disposeAllOperators` which would add "dispose exceptions" to it as suppressed.

Reply:
That's true, but what gonna happen to the suppressed exception? You still need to throw it otherwise the exception is swallowed I guess.

cc @pnowojski

Is `disposeAllOperators(true);` designed not to throw any exception just force shut down? If that's the case, I guess the best way is just logging saying that `dispose error is not the rootcause?'

Reply:
> what gonna happen to the suppressed exception?

It will be logged by
```
LOG.error("Invoke Error", invokeException);
```

Reply:
Mark this as resolved based on offline sync up.